src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect } from "react";
+import { Link } from "react-router-dom";
+
+export default function PopupWithForm({
+    setIsSignInPopup,
+    setIsSignUpPopup,
+    linkName,
+    onSubmit,
+    name,
+    title,
+    disabledButtonClass,
+    disableButton,
+    buttonText,
+    children,
+    submitError,
+    isValid
+
+
+    // isOpen,
+    // onClose, 
+    // openSigninPopup,
+    // openSignupPopup,
+}) {
+    // const fadeIn = isOpen ? "popup_fadein" : null;
+    const fadeIn = "popup__fadein"
+
+    useEffect(() => {
+        const escKeyClose = (e) => {
+            if (e.keyCode === 27) {
```

Comment:
You shouldn't use `numbers` for checking what key has been pressed because such numbers don't tell other developers what kind of key it is. `(evt.key === “Escape”)`